lib/improver/tests/wind_direction/test_WindDirection.py

```diff
@@ -0,0 +1,319 @@
+
+
+# Test the complex number handling functions.
+class Test_deg_to_complex(IrisTest):
+    """Test the deg_to_complex function."""
+
+    def test_converts_single(self):
+        """Tests that degree angle value is converted to complex."""
+        expected_out = 0.707106781187+0.707106781187j
+        result = WindDirection().deg_to_complex(45.0)
+        self.assertAlmostEqual(result, expected_out)
+
+    def test_handles_angle_wrap(self):
+        """Test that code correctly handles 360 and 0 degrees."""
+        expected_out = 1+0j
+        result = WindDirection().deg_to_complex(0)
+        self.assertAlmostEqual(result, expected_out)
+
+        expected_out = 1-0j
+        result = WindDirection().deg_to_complex(360)
+        self.assertAlmostEqual(result, expected_out)
+
+    def test_converts_array(self):
+        """Tests that array of floats is converted to complex array."""
+        result = WindDirection().deg_to_complex(DEGREE_ANGLES)
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, COMPLEX_ANGLES)
+
+
+class Test_complex_to_deg(IrisTest):
+    """Test the complex_to_deg function."""
+
+    def test_fails_if_data_is_not_array(self):
+        """Test code raises a Type Error if input data not an array."""
+        input_data = 0-1j
+        msg = ('Input data is not a numpy array, but'
+               ' {}'.format(type(input_data)))
+        with self.assertRaisesRegexp(TypeError, msg):
+            WindDirection().complex_to_deg(input_data)
+
+    def test_handles_angle_wrap(self):
+        """Test that code correctly handles 360 and 0 degrees."""
+        # Input is complex for 0 and 360 deg - both should return 0.0.
+        input_data = np.array([1+0j, 1-0j])
+        result = WindDirection().complex_to_deg(input_data)
+        self.failUnless((result == 0.0).all())
+
+    def test_converts_array(self):
+        """Tests that array of complex values are converted to degrees."""
+        result = WindDirection().complex_to_deg(COMPLEX_ANGLES)
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, DEGREE_ANGLES)
+
+
+class Test_find_r_values(IrisTest):
+    """Test the find_r_values function."""
+
+    def test_converts_single(self):
+        """Tests that r-value correctly extracted from complex value."""
+        expected_out = 2.0
+        complex_in = 1.4142135624+1.4142135624j  # Complex for angle=45 and r=2
+        result = WindDirection().find_r_values(complex_in)
+        self.assertAlmostEqual(result, expected_out)
+
+    def test_converts_array(self):
+        """Test that code can find r-values from array of complex numbers."""
+        expected_out = np.ones(COMPLEX_ANGLES.shape, dtype=np.float32)
+        result = WindDirection().find_r_values(COMPLEX_ANGLES)
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, expected_out)
+
+
+class Test_calc_polar_mean_std_dev(IrisTest):
```

Comment:
Change name to match new name, I suggest Test_calc_confidence_measure. The name in the comment below should also match.